### ISSUE TITLE
Arguments to be passed to zinbFit

### ISSUE THREAD

**Issue by drisso** [drisso/archive-zinbwave#12]
Right now, the method wants in input X, V, and the which_ arguments to know which column of X and V to pass to the internal functions.

I think this is a bit confusing for the users, so I propose to use formulas instead: something along the lines of what the pscl package does: e.g., `~ a + b | a + c` would mean that the "a" covariate is used for both mu and pi, "b" only for mu and "c" only for pi.

Need to understand a bit more how to work with formulas, but I think that model.matrix() should work here.

**Comment by drisso** [comment]
I still think that the formula interface is a good idea. However, it's slightly more complicated than the zeroinfl() case, because we have both gene-level and sample-level covariates.

We can either use two formulas, one for X and one for V or include both sample-level and gene-level covariates in the same formula. That would be the easiest for the user but possibly more confusing and the user would need to make sure not to have gene-level and sample-level covariates with the same name.

Except this complication, this should work:

``` r
zinbFit(Y, formula, ...)

## remove the | and add a +
formula[[2]][1] <- call("+")
X <- model.matrix(formula)

which_X_mu <- which(colnames(X) %in% as.character(formula[[2]][[2]]))
which_X_pi <- which(colnames(X) %in% as.character(formula[[2]][[3]]))
```

This works except for the intercept. There should be a better solution, perhaps using `model.frame`.

**Comment by jpvert** [comment]
I did not work on the formula in the new revision. zinbFit (and zinbInitialize) still take X,V etc.. as input.  But I agree that allowing a formula as you suggest would help some users, so it is still an "open issue".

**Comment by drisso** [comment]
This turned out to be much simpler than I expected and is now implemented in the summarizedexp branch (commit fe89be34f5dcde2069c7c3bc005a5c92a7770d9f).
